diff --git a/asio/include/asio/ip/impl/address.hpp b/asio/include/asio/ip/impl/address.hpp
index 56e39191..44cc2367 100644
--- a/asio/include/asio/ip/impl/address.hpp
+++ b/asio/include/asio/ip/impl/address.hpp
@@ -54,7 +54,7 @@ template <typename Elem, typename Traits>
 std::basic_ostream<Elem, Traits>& operator<<(
     std::basic_ostream<Elem, Traits>& os, const address& addr)
 {
-  return os << addr.to_string().c_str();
+  return os << addr.to_string();
 }
 
 } // namespace ip
